### PR TITLE
React warning fixes

### PR DESCRIPTION
**Pull request by martinpitt**
First wave of fixes after #19053

----

 - [x] https://github.com/cockpit-project/bots/pull/5411
 - [x] https://github.com/cockpit-project/bots/pull/5412
 - [x] https://github.com/cockpit-project/bots/pull/5413
 - [x] https://github.com/cockpit-project/bots/pull/5415

### CODE REVIEW

**Comment by martinpitt** [comment]
Oh, [this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19503-20231018-082007-fcada5d0-rhel-8-9-distropkg-other/log.html) is interesting:

> error: Tracer failed: "{"problem":null,"exit_status":1,"exit_signal":null,"message":"Traceback (most recent call last):\n  File \"<string>\", line 6, in <module>\n  File \"/usr/lib/python3.6/site-packages/tracer/query.py\", line 72, in get\n    return self._method(**self._kwargs)\n  File \"/usr/lib/python3.6/site-packages/tracer/resources/tracer.py\", line 126, in trace_affected\n    if not self._applications.find('kernel').ignore and self._has_updated_kernel():\n  File \"/usr/lib/python3.6/site-packages/tracer/resources/tracer.py\", line 134, in _has_updated_kernel\n    running = System.running_kernel_package()\n  File \"/usr/lib/python3.6/site-packages/tracer/resources/system.py\", line 119, in running_kernel_package\n    return System.package_manager().find_package(System.kernel_package_name(), os.uname()[2])\nAttributeError: 'NoneType' object has no attribute 'find_package'"}", data: """"

This should get reported and become a naughty.

The other failures are easy, fixed.

File: pkg/packagekit/updates.jsx [hunk 1]
```diff
@@ -1075,7 +1075,10 @@
                     this.setState(nextState);
                 })
                 .catch((exception, data) => {
-                    console.error(`Tracer failed: "${JSON.stringify(exception)}", data: "${JSON.stringify(data)}"`);
+                    // the common case is that this platform does not have tracer installed or polkit does not allow it
+                    if (!exception.message?.includes("ModuleNotFoundError") &&
+                        !exception.problem === "access-denied")
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply pkg/packagekit/updates.jsx]: Erk, well spotted.. Fixed

**Comment by martinpitt** [comment]
[one more common tracer exit](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19503-20231018-084853-85310b0a-centos-8-stream-pybridge-expensive/log.html#46) which we should ignore. Same thing [here](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19503-20231018-084853-85310b0a-arch-expensive/log.html#63).

**Comment by martinpitt** [comment]
We should ignore [UnsupportedDistribution as well](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19503-20231018-092727-843a74e4-arch-other/log.html#96)

File: pkg/packagekit/updates.jsx [hunk 2]
```diff
@@ -1075,7 +1075,15 @@ class OsUpdates extends React.Component {
...
-                    console.error(`Tracer failed: "${JSON.stringify(exception)}", data: "${JSON.stringify(data)}"`);
+                    // common cases: this platform does not have tracer installed
+                    if (!exception.message?.includes("ModuleNotFoundError") &&
+                        // or supported (like on Arch)
+                        !exception.message?.includes("UnsupportedDistribution") &&
+                        // or polkit does not allow it
+                        exception.problem !== "access-denied" &&
+                        // or the session goes away while checking
+                        exception.problem !== "terminated")
+                        console.error(`Tracer failed: "${JSON.stringify(exception)}", data: "${JSON.stringify(data)}"`);
```
Comment (cockpituous) [inline pkg/packagekit/updates.jsx:1086]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19503-20231018-101508-5f9f5ce3-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/shell/hosts.jsx [hunk 3]
```diff
@@ -197,7 +197,15 @@ export class CockpitHosts extends React.Component {
...
                         <PageSidebar isSidebarOpen={this.props.opened} theme="dark" className={"sidebar-hosts" + (this.state.editing ? " edit-hosts" : "")}>
-                            <CockpitNav selector={this.props.selector} groups={groups} item_render={render} sorting={(a, b) => true} filtering={this.filterHosts} current={label} />
+                            <CockpitNav
+                                selector={this.props.selector}
+                                groups={groups}
+                                item_render={render}
+                                sorting={(a, b) => true}
+                                filtering={this.filterHosts}
+                                current={label}
+                                jump={() => console.error("internal error: jump not supported in hosts selector")}
```
Comment (cockpituous) [inline pkg/shell/hosts.jsx:207]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19503-20231018-101508-5f9f5ce3-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
Meh, one more tracer bug on [RHEL 9.3](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19503-20231018-111152-5f9f5ce3-rhel-9-3-other/log.html#96) and [RHEL 9.4](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19503-20231018-111150-5f9f5ce3-rhel-9-4-other/log.html#96):

I reported this to https://issues.redhat.com/browse/RHEL-14094 , will add naughty.

File: pkg/packagekit/updates.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/packagekit/updates.jsx:1086]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19503-20231018-115234-5f9f5ce3-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/shell/hosts.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/shell/hosts.jsx:207]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19503-20231018-115234-5f9f5ce3-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]

**Pull request merged**